pipeline.py

```diff
@@ -72,31 +89,109 @@ class TextProcessingPipeline:
     """
 
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        articles = CorpusManager.get_articles(self.corpus_manager).values()
+
+        for article in articles:
+            raw_text = Article.get_raw_text(article)
+            tokens = self._process(raw_text)
+
+            cleaned = []
+            single_tagged = []
+            multiple_tagged = []
+            for token in tokens:
+                cleaned.append(token.get_cleaned())
+                single_tagged.append(token.get_single_tagged())
+                multiple_tagged.append(token.get_multiple_tagged())
+
+            article.save_as(' '.join(cleaned), ArtifactType.cleaned)
+            article.save_as(' '.join(single_tagged), ArtifactType.single_tagged)
+            article.save_as(' '.join(multiple_tagged), ArtifactType.multiple_tagged)
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        clean_text = ''
+
+        for symbol in raw_text.replace('-\n', ''):
```

Comment:
I understand why you do the replace, but other processing is compltely wrong. Just imagine how many concatenations are performed to construct a new text.

Reply:
Mystem can accept "dirty" text (with all the punctuations and other staff) and then it will process it accordingly, you only need to properly parse its output report

Reply:
done